kafka/admin/client.py

```diff
@@ -1000,22 +1002,47 @@ def _describe_consumer_groups_process_response(self, response):
-            # including KafkaAdminClient.java
-            group_description = response.groups[0]
-            error_code = group_description[0]
+            for response_field, response_name in zip(response.SCHEMA.fields, response.SCHEMA.names):
+                if type(response_field) == Array:
+                    described_groups = response.__dict__[response_name]
+                    described_groups_field_schema = response_field.array_of
+                    for described_group  in described_groups:
+                        described_group_information_list = []
+                        is_consumer_protocol_type = False
+                        for (described_group_information, group_information_name, group_information_field) in zip(described_group, described_groups_field_schema.names, described_groups_field_schema.fields):
+                            if group_information_name == 'protocol_type':
+                                protocol_type = described_group_information
+                                is_consumer_protocol_type = (protocol_type == ConsumerProtocol.PROTOCOL_TYPE or not protocol_type)
+                            if type(group_information_field) == Array:
+                                member_information_list = []
+                                member_schema = group_information_field.array_of
+                                for members in described_group_information:
+                                    member_information = []
+                                    for (member, member_field, member_name)  in zip(members, member_schema.fields, member_schema.names):
+                                        if member_name == 'member_metadata' and is_consumer_protocol_type:
+                                            member_information.append(ConsumerProtocolMemberMetadata.decode(member))
+                                        elif member_name == 'member_assignment' and is_consumer_protocol_type:
+                                            member_information.append(ConsumerProtocolMemberAssignment.decode(member))
+                                        else:
+                                            member_information.append(member)
+                                    else:
+                                        member_info_tuple = MemberInformation._make(member_information)
+                                        member_information_list.append(member_info_tuple)
+                                else:
+                                    described_group_information_list.append(member_information_list)
+                            else:
+                                described_group_information_list.append(described_group_information)
+                        else:
+                            if response.API_VERSION <=2:
+                                described_group_information_list.append([])
+                            group_description = GroupInformation._make(described_group_information_list)
```

Comment:
Agreed, that the for loop is misleading. I am removing it and hard coding it to an index of 0. The reason is that it will always return only 1 group's description. 
This is as per the java client: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java#L2867
Hope this helps. Please let me know if you need more information.

Reply:
ah, makes much more sense that way